Adds Server 2012 R2 support to xWindowsOptionalFeature

iainbrighton opened the pull request:
* Fixes #42, adding Server 2012 R2 support to xWindowsOptional feature
* Fixes bug in Test-TargetResource that always returned $true
* Adds test coverage
* Updates readme.md with xWindowsOptionalFeature

iainbrighton commented on Tests/MSFT_xWindowsOptionalFeature.Tests.ps1:
```diff
@@ -0,0 +1,311 @@
+[CmdletBinding()]
+param()
+
+Import-Module $PSScriptRoot\..\DSCResources\MSFT_xWindowsOptionalFeature\MSFT_xWindowsOptionalFeature.psm1
```
Is that what we should be using for all tests now? I just copied a file from another resource in the module!

KarolKaczmarek replied:
yes, we've added those templates recently, so majority of our modules don't use them, but that's recommended approach. Take a look at https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md

iainbrighton commented on DSCResources/MSFT_xWindowsOptionalFeature/MSFT_xWindowsOptionalFeature.psm1:
```diff
@@ -155,11 +155,11 @@ function Set-TargetResource
         else
         {
-            $feature = Dism\Enable-WindowsOptionalFeature -FeatureName $Name -Online -LogLevel $DismLogLevel @PSBoundParameters
+            $feature = Dism\Enable-WindowsOptionalFeature -FeatureName $Name -Online -LogLevel $DismLogLevel @PSBoundParameters -NoRestart
```
I thought it set this at the end of the Set-TargetResource method; at least that's what I think the existing implementation did! Will double-check.

iainbrighton commented:
@TravisEz13 @KarolKaczmarek  I've updated the xWindowsOptionalFeature tests to use the test template and moved them to the \Tests\Unit directory. Should I move the other tests too or wait until they're updated to follow the template?

iainbrighton commented:
@TravisEz13 I'm not sure I'm following the "Please consider validating the message. This could pass even if you just have a test issue" comment. Validate what message?!

iainbrighton commented:
Ah! I'm with you now. Will look at fixing that..
